### PR TITLE
GET /api/articles

### PR DESCRIPTION
**Pull request by Mpanasetckiy**
test(app.test.js): refactor test descriptions for clarity and organization
test(app.test.js): add test for GET `/api/articles/my_article` to handle bad request
test(app.test.js): add test for GET `/api/articles/777` to handle not found
test(app.test.js): add test for GET `/api/articles` to ensure correct response and data shape
test(app.test.js): add test for GET `/api/articles` with invalid query values to handle bad request
fix(articles.controllers.js): import fetchArticles function from models/articles.models.js
feat(articles.controllers.js): add getArticles function to handle GET `/api/articles` route
fix(error-handling.js): change response property name from 'msg' to 'message'
fix(routes.js): import getArticles function from controllers/articles.controllers.js
feat(routes.js): add route for GET `/api/articles`
fix(articles.models.js): import fetchArticles function from models/articles.models.js
feat(articles.models.js): add fetchArticles function to fetch articles with optional query parameters
fix(package.json): add jest-sorted package to setupFilesAfterEnv for sorting assertions in tests

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -41,36 +41,72 @@ describe("app routes", () => {
...
-        .expect(404)
-        .then({ body: "No data found" });
+    describe("getArticles", () => {
+      test("200 - GET /api/articles", () => {
+        return request(app)
+          .get("/api/articles")
+          .expect(200)
+          .then(({ body: { articles } }) => {
+            expect(articles).toHaveLength(13);
+            expect(articles).toBeSorted({
```
Comment (pippaamy) [inline __tests__/app.test.js:85]: I would have your sorted in a separate test:)

File: __tests__/app.test.js [hunk 2]
```diff
@@ -41,36 +41,72 @@ describe("app routes", () => {
...
+              key: "created_at",
+              descending: true,
+            });
+            articles.forEach((article) => {
+              expect(article).toHaveProperty("author");
+              expect(article).toHaveProperty("title");
+              expect(article).toHaveProperty("article_id");
```
Comment (pippaamy) [inline __tests__/app.test.js:92]: It would be nice to see the data type as well eg. article_id will be a number

File: __tests__/app.test.js [hunk 3]
```diff
@@ -41,36 +41,72 @@ describe("app routes", () => {
...
+              expect(article).toHaveProperty("article_img_url");
+              expect(article).toHaveProperty("comment_count");
+            });
+          });
+      });
+
+      test("400 - GET /api/articles", () => {
+        return request(app)
+          .get("/api/articles?sort_by=blahblah&order=asc")
+          .expect(400)
```
Comment (pippaamy) [inline __tests__/app.test.js:105]: not sure this is necessary yet as we arent doing a query in order to have the articles sorted by created_at. If youre going to do this test you also need to test that other queries youve put in eg testing that adding asc works

File: __tests__/app.test.js [hunk 4]
```diff
@@ -41,36 +41,72 @@ describe("app routes", () => {
...
+      });
     });
 
-    test("404 - GET /api/articles/777", () => {
-      return request(app)
-        .get("/api/articles/777")
-        .expect(404)
-        .then({ body: "No data found" });
+    describe("getArticles", () => {
+      test("200 - GET /api/articles", () => {
```
Comment (pippaamy) [inline __tests__/app.test.js:79]: can this be more descriptive?

File: controllers/articles.controllers.js [hunk 5]
```diff
@@ -11,4 +14,15 @@ const getArticleById = async (req, res, next) => {
   }
 };
 
-module.exports = { getArticleById };
+const getArticles = async (req, res, next) => {
+  try {
+    const allQueries = req.query;
+    const articles = await fetchArticles(allQueries);
+
+    res.status(200).send({ articles });
```
Comment (pippaamy) [inline controllers/articles.controllers.js:22]: this is great!

File: models/articles.models.js [hunk 6]
```diff
@@ -12,4 +12,28 @@ const fetchArticleById = async (id) => {
...
+    return Promise.reject({ status: 400, message: "Bad query value!" });
+  }
+
+  let sqlStr = `SELECT articles.author, title, articles.article_id, topic, articles.created_at, articles.votes, article_img_url, COUNT(comments)  AS comment_count
+  FROM articles
+  LEFT JOIN comments ON 
+  articles.article_id = comments.article_id
+  GROUP BY articles.article_id`;
+
+  sqlStr += ` ORDER BY ${sort_by} ${order};`;
```
Comment (pippaamy) [inline models/articles.models.js:30]: this is a great way of making it dynamic

**Review by pippaamy — COMMENTED** [review]
Great job on the routing :) If youre going to add extra parts in the model we need to test for them :) well done!

File: __tests__/app.test.js (hunk 4 above)
Comment (Mpanasetckiy) [reply __tests__/app.test.js:79]: yes thanks for the feedback, I fixed test descriptions

File: __tests__/app.test.js (hunk 1 above)
Comment (Mpanasetckiy) [reply __tests__/app.test.js:85]: yeah, I thought it would be good to separate it. I fixed it.

**Pull request merged**